src/modules/job-exec/exec_config.c

```diff
@@ -216,6 +242,69 @@ int config_init (flux_t *h, int argc, char **argv)
               exec_service,
               exec_service_override ? "enabled" : "disabled",
               sdexec_properties ? "set" : "not set");
+}
+
+/*  Initialize common configurations for use by job-exec exec modules.
+ */
+int config_init (flux_t *h, int argc, char **argv)
+{
+    const flux_conf_t *conf;
+
+    /* Per trws comment in 97421e88987535260b10d6a19551cea625f26ce4
+     *
+     * The musl libc loader doesn't actually unload objects on
+     * dlclose, so a subsequent dlopen doesn't re-clear globals and
+     * similar.
+     *
+     * So we must re-initialize globals on each config_init().
+     */
+    default_cwd = "/tmp";
+    default_job_shell = NULL;
+    default_job_shell_set_cmdline = false;
+    flux_imp_path = NULL;
+    flux_imp_path_set_cmdline = false;
+    exec_service = "rexec";
+    exec_service_set_cmdline = false;
+    exec_service_override = 0;
+    sdexec_properties = NULL;
+
+    if (!(conf = flux_get_conf (h))) {
+        flux_log_error (h, "error retrieving flux conf");
+        return -1;
+    }
+
+    if (conf_parse (h, conf) < 0)
+        return -1;
+
+    if (argv && argc) {
+        /* Finally, override values on cmdline */
+        for (int i = 0; i < argc; i++) {
+            if (strstarts (argv[i], "job-shell=")) {
+                default_job_shell = argv[i]+10;
+                default_job_shell_set_cmdline = true;
+            }
+            else if (strstarts (argv[i], "imp=")) {
+                flux_imp_path = argv[i]+4;
+                flux_imp_path_set_cmdline = true;
+            }
+            else if (strstarts (argv[i], "service=")) {
+                exec_service = argv[i]+8;
+                exec_service_set_cmdline = true;
+            }
+        }
+    }
```

Comment:
It might be a little cleaner if the command line args are re-parsed each time there is a reconfiguration rather than adding the `_set_cmdline` flags for each possible thing.

Reply:
I thought about that, but disliked "saving off" argv & argc somewhere for re-parse, since it isn't generally available at config-reload time.   Think it's a net win saving it off somewhere?

Reply:
Since everything runs within the `mod_main()` function, argc, argv remain valid and wouldn't need to be copied.